### PR TITLE
✨Source Iterable: Migrate to low code

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
Fixed https://github.com/airbytehq/airbyte-internal-issues/issues/6426

## How
- Migrated Source Iterable to Declarative Source
- Removed `AccessCheck` as we check connection with List stream
- Removed custom all_streams_accessible validation during check connection, instead used default availability logic

## Recommended reading order
1. `x.yaml`
2. `y.python`

## 🚨 User Impact 🚨
No

## Pre-merge Actions
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by lazebnyi** [comment]
I will add unit tests for the custom components.

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py [hunk 1]
```diff
@@ -2,20 +2,14 @@
...
-import requests.exceptions
-from airbyte_cdk.models import SyncMode
-from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.declarative.yaml_declarative_source import YamlDeclarativeSource
 from airbyte_cdk.sources.streams import Stream
 from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
-from source_iterable.utils import read_full_refresh
 
 from .streams import (
-    AccessCheck,
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/source_iterable/source.py:15]: Should this stream be present in the manifest?

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py [hunk 2]
```diff
@@ -53,101 +42,68 @@
...
-                    SmsReceived(authenticator=authenticator, **date_range),
-                    SmsSendSkip(authenticator=authenticator, **date_range),
-                    SmsUsageInfo(authenticator=authenticator, **date_range),
-                    Purchase(authenticator=authenticator, **date_range),
-                    CustomEvent(authenticator=authenticator, **date_range),
-                    HostedUnsubscribeClick(authenticator=authenticator, **date_range),
-                    Events(authenticator=authenticator),
-                ]
-            )
+        streams.extend(
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/source_iterable/source.py:69]: Is there a plan to move these to low code eventually? If not, why?

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py [hunk 3]
```diff
@@ -53,101 +42,68 @@
...
-            Templates(authenticator=authenticator, **date_range),
-        ]
-        # Iterable supports two types of Server-side api keys:
-        # - read only
-        # - server side
-        # The first one has a limited set of supported APIs, so others are filtered out here.
-        # A simple check is done - a read operation on a stream that can be accessed only via a Server side API key.
-        # If read is successful - other streams should be supported as well.
-        # More on this - https://support.iterable.com/hc/en-us/articles/360043464871-API-Keys-
-        if all_streams_accessible():
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/source_iterable/source.py:111]: Should we still be using this check? If not, why not?

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_extractors.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+
+@pytest.fixture
+def mock_response():
+    mock_response = Mock()
+    return mock_response
+
+@pytest.mark.parametrize("extractor_class,field_path", [(ListUsersRecordExtractor, "getUsers"), (XJsonRecordExtractor, "users"), (EventsRecordExtractor, "events")])
+def test_extract_records(mock_response, extractor_class, field_path):
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/unit_tests/test_extractors.py:23]: Nit: can you split these into 3 separate tests?

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_stream_events.py [hunk 5]
```diff
@@ -4,199 +4,72 @@
...
+from airbyte_cdk.sources.declarative.types import StreamSlice
+from source_iterable.source import SourceIterable
 
 
 @responses.activate
 @pytest.mark.parametrize(
     "response_objects,expected_objects,jsonl_body",
     [
         (
-            [
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/unit_tests/test_stream_events.py:18]: Out of curiosity, why did you remove some of these events?

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_source.py [hunk 6]
```diff
@@ -2,27 +2,17 @@
...
-def test_source_streams(mock_lists_resp, config, body, status, expected_streams):
+@pytest.mark.parametrize("body, status, expected_streams", [(b"alpha@gmail.com\nbeta@gmail.com", 200, 44)])
+def test_source_streams(config, body, status, expected_streams):
+    responses.get("https://api.iterable.com/api/lists", json={"lists": [{"id": 1}]})
     responses.add(responses.GET, "https://api.iterable.com/api/lists/getUsers?listId=1", body=body, status=status)
-    streams = SourceIterable().streams(config=config)
-    assert len(streams) == expected_streams
-
-
-def test_source_check_connection_ok(config):
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/unit_tests/test_source.py:21]: Can we patch at the http request level instead of deleting the `test_source_check_connection_*` tests?

I know that for some streams we're using the low code component, but since it looks like we're not migrating all streams it feels like a good idea to keep the coverage.

File: airbyte-integrations/connectors/source-iterable/source_iterable/manifest.yaml [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+            inject_into: header
+        error_handler:
+          type: CompositeErrorHandler
+          error_handlers:
+            - type: DefaultErrorHandler
+              response_filters:
+                - type: HttpResponseFilter
+                  action: IGNORE
+                  http_codes:
+                    - 401
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/source_iterable/manifest.yaml:94]: Given that we used to allow this stream to continue on 500s (based on the presence of the `test_listuser_stream_keep_working_on_500` test, should we have another error handler here to account for that?

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_streams.py [hunk 8]
```diff
@@ -133,138 +148,3 @@ def test_campaigns_metrics_parse_response():
...
-    slices = 0
-    for slice_ in users_stream.stream_slices(sync_mode=SyncMode.full_refresh):
-        slices += 1
-        _ = list(users_stream.read_records(stream_slice=slice_, sync_mode=SyncMode.full_refresh))
-    assert len(responses.calls) == 1
-    assert slices > 1
-
-
-@responses.activate
-def test_listuser_stream_keep_working_on_500():
```
Comment (clnoll) [inline airbyte-integrations/connectors/source-iterable/unit_tests/test_streams.py:196]: Why are we changing this behavior?

**Review by clnoll — COMMENTED** [review]
@lazebnyi I've left a few initial comments and questions for you. Let me know when it's ready for another look!

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py (hunk 1 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/source_iterable/source.py:15]: No, we don't need this stream any more. Instead this stream we use `list` stream to check connection.

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py (hunk 3 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/source_iterable/source.py:111]: No, default availability strategy will take care about that.

File: airbyte-integrations/connectors/source-iterable/source_iterable/manifest.yaml (hunk 7 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/source_iterable/manifest.yaml:94]: DEFAULT_RETRIABLE_ERRORS include 500s, so we don't need defined it in manifest

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_stream_events.py (hunk 5 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/unit_tests/test_stream_events.py:18]: Just typo.
Reverted.

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_streams.py (hunk 8 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/unit_tests/test_streams.py:196]: Updated to test backoff policy on 500 error

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_extractors.py (hunk 4 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/unit_tests/test_extractors.py:23]: Updated

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_source.py (hunk 6 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/unit_tests/test_source.py:21]: Added

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py (hunk 1 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-iterable/source_iterable/source.py:15]: 👍 Can you update the PR description with that info, and any other major changes?

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py (hunk 2 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-iterable/source_iterable/source.py:69]: Makes sense. Can you add a comment here about that?

File: airbyte-integrations/connectors/source-iterable/unit_tests/test_extractors.py (hunk 4 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-iterable/unit_tests/test_extractors.py]: Thank you!

**Review by clnoll — COMMENTED** [review]
Thanks for the updates @lazebnyi! Let me know when you've had the chance to run regression tests and I'll approve.

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py (hunk 2 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-iterable/source_iterable/source.py:71]: added

**Review by clnoll — APPROVED** [review]
